test/apps/iodemo/io_demo.cc

```diff
@@ -1366,16 +1366,17 @@ class DemoClient : public P2pDemoCommon {
         LOG << "terminate connection " << server_info.conn << " due to "
             << reason;
 
+        // Don't wait for any more completions on this connection
+        _num_sent -= get_num_uncompleted(server_info);
```

Comment:
need to make sure any completions which may arrive during the destructor progress will not be counted
maybe assert/check that "conn" is member of _server_index_lookup map when counting completion?
@evgeny-leksikov WDYT?

Reply:
I think simpler and clearer to check counting with `assert(_num_completed <= _num_sent) ` and `get_num_uncompleted(server_info) <= get_num_completed(server_info) ` in `DemoClient::handle_operation_completion()`, check in lookup map could be additional to make sure we handle structures correctly

Reply:
ok, added asserts

Reply:
move `delete server_info.conn` before this line